aws/ec2metadata/api.go

```diff
@@ -20,6 +20,20 @@ func (c *EC2Metadata) GetMetadata(p string) (string, error) {
 	return output.Content, req.Send()
 }
 
+// GetDynamicData uses the path provided to request
+func (c *EC2Metadata) GetDynamicData(p string) (string, error) {
+	op := &request.Operation{
+		Name:       "GetMetadata",
+		HTTPMethod: "GET",
+		HTTPPath:   path.Join("/", "dynamic", p),
+	}
+
+	output := &metadataOutput{}
+	req := c.NewRequest(op, nil, output)
+
+	return output.Content, req.Send()
```

Comment:
Well, the problem here is that instance identity doc has different base path and cannot be obtained via `GetMetadata` since `GetMetadata` effectively calls `http://169.254.169.254/meta-data` whereas instance identity doc is available under `http://169.254.169.254/dynamic`. That is why I create this extra function.

Should I create another level of abstraction and built `GetDynamicData` & `GetMetadata` on top of it?